src/api/sovereignVenue.ts

```diff
@@ -0,0 +1,27 @@
+import firebase from "firebase/app";
+
+export const fetchSovereignVenueId = async (
+  venueId: string
+): Promise<string> => {
+  const venueDoc = await firebase
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .get();
+
+  if (!venueDoc.exists) {
+    throw new Error("The venue doesn't exist");
+  }
+
+  const venue = venueDoc.data();
+
+  if (!venue || !venueDoc.id) {
+    throw new Error("The venue doesn't have data or id");
+  }
+
+  if (!venue.parentId) {
+    return venueDoc.id;
+  } else {
+    return fetchSovereignVenueId(venue.parentId);
+  }
```

Comment:
There is a potential for this to turn into an infinite loop if there is a circular reference in the 'parent venue chain'.

We should probably keep track of the venues we've already checked, and if it's one of them, give up/throw an error/return the 'highest' venue we found/similar.